src/platforms/android/androidutils.cpp

```diff
@@ -134,3 +135,73 @@ void AndroidUtils::appReviewRequested() {
       "org/mozilla/firefox/vpn/qt/VPNAppReview", "appReviewRequested",
       "(Landroid/app/Activity;)V", activity.object<jobject>());
 }
+
+// static
+void AndroidUtils::dispatchToMainThread(std::function<void()> callback) {
+  QTimer* timer = new QTimer();
+  timer->moveToThread(qApp->thread());
+  timer->setSingleShot(true);
+  QObject::connect(timer, &QTimer::timeout, [=]() {
+    callback();
+    timer->deleteLater();
+  });
+  QMetaObject::invokeMethod(timer, "start", Qt::QueuedConnection);
+}
+
+// static
+QByteArray AndroidUtils::getQByteArrayFromJString(JNIEnv* env, jstring data) {
+  const char* buffer = env->GetStringUTFChars(data, nullptr);
+  if (!buffer) {
+    logger.error() << "getQByteArrayFromJString - failed to parse data.";
+    return QByteArray();
+  }
+  QByteArray out(buffer);
+  env->ReleaseStringUTFChars(data, buffer);
+  return out;
+}
+
+// static
+QString AndroidUtils::getQStringFromJString(JNIEnv* env, jstring data) {
+  const char* buffer = env->GetStringUTFChars(data, nullptr);
+  if (!buffer) {
+    logger.error() << "getQStringFromJString - failed to parse data.";
+    return QString();
+  }
+  QString out(buffer);
+  env->ReleaseStringUTFChars(data, buffer);
+  return out;
+}
+
+// static
+QJsonObject AndroidUtils::getQJsonObjectFromJString(JNIEnv* env, jstring data) {
+  QJsonObject empty = QJsonObject();
```

Comment:
You use this only as the fallback value. You can also remove this line, and when needed, simply call: `return QJsonObject();`.